### PR TITLE
Task/ug feature update

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR cleans up and updates the feature sections of the user guide to be more complementary to the recent tutorial content rework.
- This is a draft PR. There is more work to do.

The generated docs for this PR are here: https://raja.readthedocs.io/en/task-ug-feature-update/sphinx/user_guide/features.html

### CODE REVIEW

**Review by artv3 — APPROVED** [review]

File: docs/sphinx/user_guide/feature/loop_basic.rst
Comment (MrBurmark) [inline docs/sphinx/user_guide/feature/loop_basic.rst:106]: typo, `loop nest.`?

File: docs/sphinx/user_guide/feature/loop_basic.rst
Comment (MrBurmark) [inline docs/sphinx/user_guide/feature/loop_basic.rst:199]: typo, spacing

File: docs/sphinx/user_guide/feature/resource.rst
Comment (mdavis36) [inline docs/sphinx/user_guide/feature/resource.rst:268]:
```suggestion
and operations, as well as define and control asynchronous execution patterns.
```

File: docs/sphinx/user_guide/feature/resource.rst
Comment (mdavis36) [inline docs/sphinx/user_guide/feature/resource.rst:343]: We don't need a wait call as the memcpy is enqueued on the same stream as the final raja kernel. If the memcpy was enqueued on `res_gpu2` we would need a wait

File: docs/sphinx/user_guide/feature/loop_basic.rst
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/loop_basic.rst:199]: fixed

File: docs/sphinx/user_guide/feature/resource.rst
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/resource.rst:179]: fixed

File: docs/sphinx/user_guide/feature/resource.rst
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/resource.rst:268]: changed

File: docs/sphinx/user_guide/feature/resource.rst
Comment (rhornung67) [reply docs/sphinx/user_guide/feature/resource.rst:343]: comment removed.

**Comment by rhornung67** [comment]
@MrBurmark @artv3 @mdavis36 @rchen20 this should be good to merge after your reviews and approvals. Please note that I will have one more doc update PR coming soon to get merged before the release.

**Comment by rhornung67** [comment]
@MrBurmark @artv3 @rchen20 @mdavis36 I've addressed all the comments on this PR that you provided. Any issues or concerns with me merging it?

**Comment by MrBurmark** [comment]
I'm not concerned at all.

**Review by mdavis36 — APPROVED** [review]

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**
